textual_fspicker/base_dialog.py

```diff
@@ -100,11 +112,14 @@ def compose(self) -> ComposeResult:
         """
         with Dialog() as dialog:
             dialog.border_title = self._title
+            currentDrive = str(os.path.splitdrive(MakePath.of(self._location).expanduser().absolute())[0]+'\\').upper()
```

Comment:
This library is written so that it uses `pathlib` (and so also allows anyone to provide a `Path`-derived class so other implementations can be used). Could you consider a version of what you're doing here that follows the same approach.